submissions/sung/bonus-testing-part-two.js

```diff
@@ -0,0 +1,88 @@
+*/
+
+
+/* As a refresher, presented again is a basic example of a test
+    "framework" where our test is a function that we can reuse
+    to check against different components of our application.
+    The `!expression` may look odd - we test for failing
+    functionality first to then refactor the test to pass.
+    This is common practice in testing ('red-green' refactoring).
+*/
+function expect(expression, failureMessage, successMessage) {
+  if (!expression) {
+    console.log('test failed:', failureMessage);
+    return;
+  }
+  console.log('test passed:', successMessage);
+}
+
+/* Below is an example of our test in action. Run this file in node
+    to see what happens when it fails, and change `ricksFaveAnimal`
+    to get it to pass!
+*/
+var ricksFaveAnimal = 'hyena';
+
+expect(
+  ricksFaveAnimal === 'penguin',
+  'ricksFavoriteAnimal should equal penguin, but currently equals ' + ricksFaveAnimal,
+  'ricksFavoriteAnimal equals penguin!');
+
+  // BEGIN WORK BELOW - test code by running `node bonus-testing-part-two.js`
+  //  in your terminal!
+
+  /* ===================================================================
+  ---------------------- Part Two! Hungry Lion ----------------------------
+  ======================================================================
+   As long as the lion is well-fed, he doesn't care too much of the
+   humans that pass through. Unfortunately, the new caretaker is a little
+   absent-minded.
+
+   The lion needs 4 meals per day on average to stay happy. You're going to
+   figure out how many days it took before the lion decided to supplement his
+   diet with the caretaker.
+  */
+
+  // number of times the new caretaker fed the lion. one array entry per day
+var mealsPerDay = [5, 4, 3, 6, 2, 4, 3, 4, 5, 1];
+var tooHungryDay;
+
+  /*
+   TODO:
+   Cycle through the days in mealsPerDay. At each day, print out the average
+   number of meals/day the lion got since the new caretaker started.
+   tooHungryDay should receive the number of days before the lion started
+   pondering protein supplements (the first day the average dips below 4
+   meals)
+  */
+var averageMealsPerDay =  mealsPerDay.map(function(meal, index, array) {
+  var average = 0;
+  for (var i = 0; i < index + 1; i++) {
+    average += array[i] / (index + 1);
+  }
+  console.log(average);
+  return average;
+});
+
+tooHungryDay = 1 + averageMealsPerDay.findIndex(function(day) {
+  return day < 4;
+});
+
+
+expect(
+  typeof(tooHungryDay) === 'number',
+  'tooHungryDay should be a number but instead is a data type of ' + typeof(tooHungryDay),
+  'The lion appears to be too hungry after ' + tooHungryDay + ' days...');
+
+  // TODO:
+  // Write a second test expecting that tooHungryDay falls within an acceptable answer
+  // based on the number of days available in the array. Remember to:
+  // pass in your expression, and write a failure and a success message.
+expect(
+  0 < tooHungryDay < mealsPerDay.length,
```

Comment:
we discussed this in lecture, but let me know if you have any questions about it,
i wish we could chain booleans like this :), can do these by &&ing two boolean tests
like 0 < tooHungryDay && toHungerDay < mealsPerDay.length